extensions/interactions/ItemSelectionInput/validator.js

```diff
@@ -89,15 +89,60 @@ oppia.filter('oppiaInteractiveItemSelectionInputValidator', [
         baseInteractionValidationService.getAnswerGroupWarnings(
           answerGroups, stateName));
 
-      if (!defaultOutcome ||
-          $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+      var selectedChoices = [];
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              var choicePreviouslySelected = (
+                selectedChoices.indexOf(ruleSpecs[j].inputs.x[k]) !== -1);
+              if (!choicePreviouslySelected) {
+                selectedChoices.push(ruleSpecs[j].inputs.x[k]);
+              }
+            }
+          }
+        }
+      }
+      if (selectedChoices.length < numChoices) {
+        if (!defaultOutcome ||
+            $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+          warningsList.push({
+            type: WARNING_TYPES.ERROR,
+            message: (
+              'Please clarify the default outcome so it is less confusing to ' +
+              'the user.')
+          });
+        }
+      }
+      // For situations where max choices is 1,
+      // ensure that number of answers matches number of choices
+      if (maxAllowedCount === 1 && answerGroups.length > numChoices) {
         warningsList.push({
           type: WARNING_TYPES.ERROR,
           message: (
-            'Please clarify the default outcome so it is less confusing to ' +
-            'the user.')
+            'Please ensure that the number of choices and answers match.')
         });
       }
+      // For situations where max choices is 1,
+      // ensure that containsAtLeastOneOf only handles one option
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf' &&
```

Comment:
Okay. I must have misunderstood your previous comment. My understanding of what you said:

> Unlike multiple choice, the item selection interaction has three types of rules, which you can find in extensions/interactions/rules.json. They are: Equals, ContainsAtLeastOneOf, and DoesNotContainAtLeastOneOf (which should really be 'omits at least one of').

For the case of single-choice, these translate to:

"is the same as the given choice" (which must only have one item ticked)
"is the same as the given choice" (ditto)
"is not any of the given choices" (this rule may have any number of items ticked).
> 

Was that for contains one and equal, they should only cover one answer choice. Is that wrong?